ledger-service/http-json/src/main/scala/com/digitalasset/http/HttpService.scala

```diff
@@ -260,14 +259,31 @@ object HttpService {
     bindingEt.run: Future[Error \/ (ServerBinding, Option[ContractDao])]
   }
 
-  private[http] def httpsConnectionContext(config: TlsConfiguration): HttpsConnectionContext = {
-    import io.netty.buffer.ByteBufAllocator
-    val sslContext =
-      config.server
-        .getOrElse(
-          throw new IllegalArgumentException(s"$config could not be built as a server ssl context")
-        )
-    ConnectionContext.httpsServer(() => sslContext.newEngine(ByteBufAllocator.DEFAULT))
+  private[http] def httpsConnectionContext(config: HttpsConfig): HttpsConnectionContext = {
+    import java.io.FileInputStream
+    import java.security.{KeyStore, SecureRandom}
+    import javax.net.ssl.{SSLContext, KeyManagerFactory, TrustManagerFactory}
+
+    val keyStoreFile = config.keyStoreFile
+    val password = config.keyStorePassword.toCharArray
+
+    val keyStore = KeyStore.getInstance("PKCS12")
```

Comment:
While I think it is robust idea to use the keystore approach here. I would advocate for creating KeyStore by hand using PEM and CRT files rather than a serialized store p12.
It creates a cognitive dissonance because nowhere else in our stack do we use p12 files

Reply:
I see, yes, that makes sense. Let me try that.

Reply:
Done. I somehow hadn't seen how you programmatically build the KeyStore before.

Reply:
I looked at how you could do import private key directly into a key store and it seemed that all avenues had some obstacle in one form of another. 
I am glad you have managed to find a way through all these APIs that always miss the single call that you need the most